packages/flutter_tools/test/general.shard/project_file_invalidator_test.dart

```diff
@@ -3,67 +3,78 @@
 // found in the LICENSE file.
 
 import 'package:file/memory.dart';
-import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
 import 'package:flutter_tools/src/run_hot.dart';
-import 'package:meta/meta.dart';
 
 import '../src/common.dart';
-import '../src/context.dart';
 
 // assumption: tests have a timeout less than 100 days
 final DateTime inFuture = DateTime.now().add(const Duration(days: 100));
+final DateTime inPast = DateTime.now().subtract(const Duration(days: 100));
 
 void main() {
-  group('ProjectFileInvalidator', () {
-    _testProjectFileInvalidator(asyncScanning: false);
-  });
-  group('ProjectFileInvalidator (async scanning)', () {
-    _testProjectFileInvalidator(asyncScanning: true);
+  FakePlatform platform;
+  BufferLogger logger;
+
+  setUp(() {
+    platform = FakePlatform();
+    logger = BufferLogger();
   });
-}
 
-void _testProjectFileInvalidator({@required bool asyncScanning}) {
-  const ProjectFileInvalidator projectFileInvalidator = ProjectFileInvalidator();
+  for (bool asyncScanning in <bool>[true, false]) {
+    test('No last compile, asyncScanning: $asyncScanning', () async {
```

Comment:
These probably need to use testUsingContext() with a context that maps everything to a fake implementation that crashes or a Mock with no behavior to ensure that, e.g. some library called by the `ProjectFileInvalidator` now or in the future doesn't use default globals, like for the file system:

https://github.com/flutter/flutter/blob/948e2b010126d9f03b590a636a8af01b797c2b0f/packages/flutter_tools/lib/src/base/file_system.dart#L25

Reply:
We wrapper in a test context, possible fiddle with context a bit so that it threw if accessed at all? I think that would be preferable to creating a bunch of throwing mocks

Reply:
sgtm

Reply:
Giving this a shot by wrapping the existing `test` method ... we'll see if this works. else we could name it `testWithoutContext` ... or hopefully something shorter